query-core/src/main/java/org/infinispan/query/core/impl/MetadataHybridQuery.java

```diff
@@ -54,7 +54,15 @@ protected ClosableIteratorWithCount<ObjectFilter.FilterResult> getInternalIterat
    }
 
    private ObjectFilter.FilterResult filter(EntityEntry<Object, S> entry) {
-      ObjectFilter.FilterResult filter = objectFilter.filter(entry.key(), entry.value(), entry.metadata());
+      S value = entry.value();
```

Comment:
Don't you need to fix `HybridQuery` too? Or does it make sense to fix this in the `ObjectFilter` by rejecting anything with `value == null`?

Reply:
Yeah, it isn't quite clear which cases fetch the Cache and when others don't.

Been discussing with @fax4ever directly, but I also worry that cases when queries do access the Cache they would be refreshing last access times for things like eviction and expiration, which may or may not be desirable. From a consistency standpoint I would think is undesirable.

Reply:
Glad I'm not the only one confused :shushing_face: 

hmm... I think it is ok to update the access time (eviction and expiration). I bet some projects use only the query API and not `cache.get()` :smiley: 


We may introduce flags to control the behavior (if users ask for it)
* Update the access time (expiration)
* Update the eviction position
* Keep the entry in memory if it is loaded from the cache-loader

Reply:
the validity (consistency) we can guarantee during the asynchronous moment in which the index still has a reference for an expired entity is limited (also in the literature!).
I think that for the answer we return during the asynchronous moment could be acceptable to count or not all the contribution.